src/sandstorm/seccomp-bpf/gen-clean-h.c

```diff
@@ -0,0 +1,152 @@
+// This program prints out a "clean" header for use in bpf assembly, defining
+// constants we need from various system headers. `bpf_asm` will choke on the
+// originals, for two reasons:
+//
+// 1. They contain C code
+// 2. The #defines use expressions, which `bpf_asm` doesn't understand.
```

Comment:
In the Workers Runtime I have a hack that actually finds parenthesized expressions and evaluates them using bash's `$(())`...

```
EXPRESSION_REGEX="^([^(]*)\((.*)\)([^)]*)$"

# ... later on, in the line-reading loop ...
      elif [[ "$LINE" =~ $EXPRESSION_REGEX ]]; then
        PREFIX=${BASH_REMATCH[1]}
        EXPRESSION=${BASH_REMATCH[2]}
        SUFFIX=${BASH_REMATCH[3]}

        # Remove unsigned/long suffixes.
        EXPRESSION=$(echo "$EXPRESSION" | tr -d UuLl)

        VALUE=$(( EXPRESSION ))

        echo "$PREFIX$VALUE$SUFFIX"
```

It mostly works, believe it or not.

Reply:
You don't have to use either of these suggestions though, if you'd rather keep it this way that's up to you. Just wanted to bring up what worked for me.

Reply:
I think I'll stick with the existing approach.